.github/workflows/update.yml

```diff
@@ -11,42 +11,88 @@ jobs:
+                  pr.head.ref.startsWith(branchPrefix)
+                  && (pr.head.repo.full_name === repoFullName)
+                ))
 
-      - name: Update cucumber tag
-        id: cucumber
-        run: bin/update_cucumber
+                if (matchingPR) {
+                  return matchingPR
+                }
+              }
+
+              // No existing PRs
+              return false
+            }
 
-      - name: Re-install dependencies
-        run: composer update
-        if: steps.cucumber.outputs.cucumber_version
+            const branchPrefix = 'cucumber-update-'
 
-      - name: Update translations
-        run: bin/update_i18n
-        if: steps.cucumber.outputs.cucumber_version
+            const pr = await findOpenPR({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              branchPrefix
+            })
 
-      - name: Find changelog
-        id: changelog
-        run: bin/cucumber_changelog ${{ steps.cucumber.outputs.cucumber_version }}
-        if: steps.cucumber.outputs.cucumber_version
+            if (pr) {
+              core.info(`Found existing PR ${pr.number} for ${pr.head.ref}`)
+              core.setOutput('exists', 'true')
+              core.setOutput('checkout_branch', pr.head.ref)
+              core.setOutput('commit_branch', pr.head.ref)
+            } else {
+              const timestamp = Math.floor(Date.now() / 1000)
+              const newBranch = `do-update-${timestamp}`
 
-      - name: Open a PR
-        uses: peter-evans/create-pull-request@v7
-        if: steps.cucumber.outputs.cucumber_version
+              core.info(`No existing PR - starting new branch ${newBranch}`)
+              core.setOutput('exists', 'false')
+              core.setOutput('checkout_branch', 'master')
+              core.setOutput('commit_branch', newBranch)
+            }
+
+      - uses: actions/checkout@v4
         with:
-          commit-message: Automatic Cucumber tag update to ${{ steps.cucumber.outputs.cucumber_version }}
-          branch: cucumber-update-${{ steps.cucumber.outputs.cucumber_version }}
-          delete-branch: true
-          title: Cucumber update ${{ steps.cucumber.outputs.cucumber_version }}
-          body: ${{ steps.changelog.outputs.changelog }}
-          base: 'master'
+          ref: ${{ steps.check-existing-pr.checkout_branch }}
+
+      - run: bin/update_cucumber
+        id: cucumber
+
+      - run: composer update
+        if: steps.cucumber.outputs.cucumber_updated == 'yes'
+
+      - run: bin/update_i18n
+        if: steps.cucumber.outputs.cucumber_updated == 'yes'
+
+      - uses: actions/create-github-app-token@v2
+        id: app-token
+        if: steps.cucumber.outputs.cucumber_updated == 'yes'
+        with:
+          app-id: ${{ vars.GHERKIN_UPDATER_APP_ID }}
+          private-key: ${{ secrets.GHERKIN_UPDATER_APP_KEY }}
+
+      - run: .github/commit-and-push-gherkin-update.sh
```

Comment:
I don't want to use the current `peter-evans/create-pull-request` action because that is explicitly designed to always create a single new commit off the main branch and force push it. If there are multiple gherkin releases before we get round to merging the branch, I would much prefer to see these as separate updates each with their own build results etc.

There are other "auto commit and push" actions, but they all have limitations / design decisions that made them not quite right for this. For example, they often use the github token provided in the initial `actions/checkout` step - but in our case it's cleaner and more secure to only create the custom app token right at the end, if we know we have something to push.